pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/plan/OpChainExecutionContext.java

```diff
@@ -62,10 +65,10 @@ public OpChainExecutionContext(MailboxService mailboxService, long requestId, in
     _traceEnabled = traceEnabled;
   }
 
-  public OpChainExecutionContext(PhysicalPlanContext physicalPlanContext) {
+  public OpChainExecutionContext(PhysicalPlanContext physicalPlanContext, Executor executor) {
     this(physicalPlanContext.getMailboxService(), physicalPlanContext.getRequestId(), physicalPlanContext.getStageId(),
         physicalPlanContext.getServer(), physicalPlanContext.getDeadlineMs(), physicalPlanContext.getStageMetadata(),
-        physicalPlanContext.getPipelineBreakerResult(), physicalPlanContext.isTraceEnabled());
+        physicalPlanContext.getPipelineBreakerResult(), physicalPlanContext.isTraceEnabled(), executor);
```

Comment:
nit: it was wondering if we should separate the plan context from the machine context. 
based on what i see: `mailboxService` / `executor` is machine context and the rest of physicalPlanContext should still remains. 

This should be a good start but we can do better in next iteration